src/main/java/uk/gov/hmcts/divorce/caseworker/service/EmailUpdateService.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.divorce.caseworker.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.divorce.caseworker.service.notification.GeneralEmailNotification;
+import uk.gov.hmcts.divorce.caseworker.service.task.*;
+import uk.gov.hmcts.divorce.common.notification.EmailUpdatedNotification;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.document.task.DivorceApplicationRemover;
+import uk.gov.hmcts.divorce.systemupdate.service.task.GenerateD84Form;
+
+import static uk.gov.hmcts.divorce.divorcecase.task.CaseTaskRunner.caseTasks;
+
+@Service
+@Slf4j
+public class EmailUpdateService {
+
+    @Autowired
+    private SendCaseInviteToApplicant1 sendCaseInviteToApplicant1;
+    @Autowired
+    private SendCaseInviteToApplicant2 sendCaseInviteToApplicant2;
+    @Autowired
+    private SetCaseInviteApplicant1 setCaseInviteApplicant1;
+    @Autowired
+    private SetCaseInviteApplicant1 setCaseInviteApplicant2;
+    @Autowired
+    private EmailUpdatedNotification emailUpdatedNotification;
+    public CaseDetails<CaseData, State> processUpdateForApplicant1(final CaseDetails<CaseData, State> caseDetails) {
+        return caseTasks(
+            setCaseInviteApplicant1,
+            sendCaseInviteToApplicant1
```

Comment:
is there a need to have this as separate tasks?
I think I just had the set and send together in the aboutToSubmit

Reply:
I think the POC has a data.setDueDate(LocalDate.now().plus(2, ChronoUnit.WEEKS));
I haven't seen that on the PR yet, think it's to make sure folk wil only be able to use the access code if within 2 weeks - it's what we have in place already on the invite app2.